### PR TITLE
[SPARK-22223][SQL] ObjectHashAggregate should not introduce unnecessary shuffle

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

`ObjectHashAggregateExec` should override `outputPartitioning` in order to avoid unnecessary shuffle.

## How was this patch tested?

Added Jenkins test.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by viirya** [comment]
cc @cloud-fan @gatorsmile @HyukjinKwon

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 1]
```diff
@@ -636,4 +637,33 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
...
+
+      val sortAggPlans = aggPlan.collect {
+        case sortAgg: SortAggregateExec => sortAgg
+      }
+      assert(sortAggPlans.isEmpty)
+
+      val objHashAggPlans = aggPlan.collect {
+        case objHashAgg: ObjectHashAggregateExec => objHashAgg
+      }
+      assert(objHashAggPlans.length > 0)
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala]: Not a big deal at all but maybe `nonEmpty`?

**Review by HyukjinKwon — APPROVED** [review]
LGTM. Just double checked that this is the last one left implementing `UnaryExecNode`.

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala (hunk 1 above)
Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala]: Ok. Sounds good.

**Comment by cloud-fan** [comment]
LGTM

**Comment by viirya** [comment]
Thanks @HyukjinKwon @cloud-fan

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**